### PR TITLE
CORE-4423: Add avro changes for node state and update

### PR DESCRIPTION
**Pull request by benmcmahon100**
Corresponding `corda-runtime-os` PR: https://github.com/corda/corda-runtime-os/pull/1636

### CODE REVIEW

File: data/avro-schema/src/main/resources/avro/net/corda/data/virtualnode/VirtualNodeStateChangeRequest.avsc [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "type": "record",
+  "name": "VirtualNodeStateChangeRequest",
+  "namespace": "net.corda.data.virtualnode",
+  "fields": [
+    {
+      "name": "holdingIdentityShortHash",
```
Comment (vkolomeyko) [inline data/avro-schema/src/main/resources/avro/net/corda/data/virtualnode/VirtualNodeStateChangeRequest.avsc:7]: Should we be using:
```
    {
      "name": "holdingIdentity",
      "type": "net.corda.data.identity.HoldingIdentity"
    }
```
to identify vNode?

File: data/avro-schema/src/main/resources/avro/net/corda/data/virtualnode/VirtualNodeStateChangeResponse.avsc [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "type": "record",
+  "name": "VirtualNodeStateChangeResponse",
+  "namespace": "net.corda.data.virtualnode",
+  "fields": [
+    {
+      "name": "holdingIdentityShortHash",
```
Comment (vkolomeyko) [inline data/avro-schema/src/main/resources/avro/net/corda/data/virtualnode/VirtualNodeStateChangeResponse.avsc:7]: Ditto about `net.corda.data.identity.HoldingIdentity` ?

File: gradle.properties [hunk 3]
```diff
@@ -7,7 +7,7 @@ cordaProductVersion = 5.0.0
 # NOTE: update this each time this module contains a breaking change
 ## NOTE: currently this is a top level revision, so all API versions will line up, but this could be moved to
 ##   a per module property in which case module versions can change independently.
-cordaApiRevision = 141
```
Comment (vkolomeyko) [inline gradle.properties]: I think the version has moved on now.

File: data/avro-schema/src/main/resources/avro/net/corda/data/virtualnode/VirtualNodeStateChangeRequest.avsc (hunk 1 above)
Comment (benmcmahon100) [reply data/avro-schema/src/main/resources/avro/net/corda/data/virtualnode/VirtualNodeStateChangeRequest.avsc:7]: The short hash is enough to identify/lookup a virtual node instance, but can change if you feel strongly on it?

File: gradle.properties (hunk 3 above)
Comment (benmcmahon100) [reply gradle.properties]: Will rebase and catchup back up soon - I think there were ~3 bumps yesterday

File: data/avro-schema/src/main/resources/avro/net/corda/data/virtualnode/VirtualNodeStateChangeRequest.avsc (hunk 1 above)
Comment (benmcmahon100) [reply data/avro-schema/src/main/resources/avro/net/corda/data/virtualnode/VirtualNodeStateChangeRequest.avsc:7]: Relevant runtime snippet:
image

File: data/avro-schema/src/main/resources/avro/net/corda/data/virtualnode/VirtualNodeInfo.avsc [hunk 4]
```diff
@@ -11,6 +11,7 @@
       "name": "cpiIdentifier",
       "type": "net.corda.data.packaging.CpiIdentifier"
     },
+
```
Comment (sprintengine) [inline data/avro-schema/src/main/resources/avro/net/corda/data/virtualnode/VirtualNodeInfo.avsc:14]: Can remove this extra line

Comment (benmcmahon100) [reply data/avro-schema/src/main/resources/avro/net/corda/data/virtualnode/VirtualNodeInfo.avsc:14]: Oops!

File: data/avro-schema/src/main/resources/avro/net/corda/data/virtualnode/VirtualNodeStateChangeRequest.avsc [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      "type": "string"
+    },
+    {
+      "name": "cpiId",
```
Comment (sprintengine) [inline data/avro-schema/src/main/resources/avro/net/corda/data/virtualnode/VirtualNodeStateChangeRequest.avsc]: Can remove this if we're sticking with identifying by holding identity

File: data/avro-schema/src/main/resources/avro/net/corda/data/virtualnode/VirtualNodeCreateResponse.avsc [hunk 6]
```diff
@@ -83,6 +83,11 @@
         "string"
       ],
       "doc": "ID of HSM connection. Null value means that HSM is not used."
+    },
+    {
+      "name": "virtualNodeState",
+      "type": "string",
+      "doc": "The current maintenance state of the virtual node"
```
Comment (driessamyn) [inline data/avro-schema/src/main/resources/avro/net/corda/data/virtualnode/VirtualNodeCreateResponse.avsc]:
```suggestion
      "doc": "The current maintenance state of the virtual node."
```

File: data/avro-schema/src/main/resources/avro/net/corda/data/virtualnode/VirtualNodeInfo.avsc [hunk 7]
```diff
@@ -36,6 +37,11 @@
       "type": ["null", "string"],
       "doc": "ID of HSM connection. Null value means that HSM is not used."
     },
+    {
+      "name": "virtualNodeState",
+      "type": "string",
+      "doc": "The current maintenance state of the virtual node"
```
Comment (driessamyn) [inline data/avro-schema/src/main/resources/avro/net/corda/data/virtualnode/VirtualNodeInfo.avsc]:
```suggestion
      "doc": "The current maintenance state of the virtual node."
```

**Review by driessamyn — COMMENTED** [review]
nitpicky

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Pull request merged**